packages/eslint-plugin/src/rules/no-magic-numbers.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * @fileoverview Rule to flag statements that use magic numbers (adapted from https://github.com/danielstjules/buddy.js)
+ * @author Scott O'Hara
+ */
+
+import { TSESTree, AST_NODE_TYPES } from '@typescript-eslint/typescript-estree';
+import baseRule from 'eslint/lib/rules/no-magic-numbers';
+import * as util from '../util';
+import { JSONSchema4 } from 'json-schema';
+
+type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
+type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+
+// Extend base schema with additional property to ignore TS numeric literal types
+const properties = {
+  ...(baseRule.meta.schema as JSONSchema4[])[0].properties,
+  ignoreNumericLiteralTypes: {
+    type: 'boolean',
+  },
+};
+
+export default util.createRule<Options, MessageIds>({
+  name: 'no-magic-numbers',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallow magic numbers',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    schema: { ...baseRule.meta.schema, ...properties },
```

Comment:
You've changed the type of the schema with this change, it should be an array.
We should probably create a utility to make this easier in future.

```suggestion
const baseRuleSchema = (baseRule.meta.schema as JSONSchema4[])[0];
// ....


    // Extend base schema with additional property to ignore TS numeric literal types
    schema: [{
      ...baseRuleSchema,
      properties: {
        ...baseRuleSchema.properties,
        ignoreNumericLiteralTypes: {
          type: 'boolean',
        },
      }
    }],
```

Reply:
Done.